### PR TITLE
Redirect missing 404 requests to a proper 404 page on the frontend

### LINKED ISSUE #912: Redirect missing 404 requests to a proper 404 page on the frontend
If you go to saveourtool.com/#/hui you should be redirected to 404 page.
Same with projects, with executions and ALL pages that are missing.

### PR DESCRIPTION
**Pull request by Cheshiriks**
* Added redirect for 404 requests
Closes #912

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt [hunk 1]
```diff
@@ -46,18 +47,30 @@ val requestModalHandler: FC<PropsWithChildren> = FC { props ->
...
                 errorLabel = response?.status.toString(),
+                status = response?.status,
             )
         }
         setModalState(newModalState)
     }
 
     modal { modalProps ->
+        if (modalState.status == 404.toShort()) {
+            window.location.href = "${window.location.origin}/error"
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt:72]: We shouldn't redirect to the generic fallback page. We should clearly indicate that it's a 404 Not Found situation

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt [hunk 2]
```diff
@@ -46,18 +47,30 @@ val requestModalHandler: FC<PropsWithChildren> = FC { props ->
...
                 isErrorModalOpen = response != null,
                 errorMessage = "${response?.status} ${response?.statusText}",
                 errorLabel = response?.status.toString(),
+                status = response?.status,
             )
         }
         setModalState(newModalState)
     }
 
     modal { modalProps ->
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt:70]: I had in mind something more like
```kotlin
if (modalState.status == 404.toShort()) {
    Redirect { to = ... }
} else {
    modal {}
}
```
For me the current version (when location is changed inside rendering of `modal`) looks like an unexpected side effect

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/App.kt [hunk 3]
```diff
@@ -161,28 +161,28 @@ class App : ComponentWithScope<PropsWithChildren, AppState>() {
                                 }
 
                                 Route {
-                                    path = "/:user/settings/profile"
+                                    path = "/${state.userInfo?.name}/settings/profile"
```
[the remark this answers is not in the document]
Comment (petertrr) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/App.kt:164]: Ok, makes sense. However, does that mean that the page `/null/settings/profile` will be present for anonymous users?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt]: Yes, agree

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt]: Changed it

**Review by petertrr — APPROVED** [review]

**Pull request merged**